### PR TITLE
Chore: Move subnet & Security Group providers from cloudprovider dependency

### PR DESCRIPTION
**Pull request by engedaam**
Fixes # 

**Description**
- Moved subnet and security group providers into a provider folder to help with dependency cycles with cloudprovider

**How was this change tested?**
- Ran all tests 

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/providers/securitygroup/provider.go [hunk 1]
```diff
@@ -105,10 +105,16 @@ func (p *SecurityGroupProvider) getSecurityGroups(ctx context.Context, filters [
...
-func (p *SecurityGroupProvider) securityGroupIds(securityGroups []*ec2.SecurityGroup) []string {
+func (p *Provider) securityGroupIds(securityGroups []*ec2.SecurityGroup) []string {
 	names := []string{}
 	for _, securityGroup := range securityGroups {
 		names = append(names, aws.StringValue(securityGroup.GroupId))
 	}
 	return names
 }
+
+func (p *Provider) ResetCache() {
```
Comment (jonathan-innis) [inline pkg/providers/securitygroup/provider.go:116]:
```suggestion
func (p *Provider) Reset() {
```

File: pkg/providers/subnet/provider.go [hunk 2]
```diff
@@ -113,3 +113,9 @@ func prettySubnets(subnets []*ec2.Subnet) []string {
 	}
 	return names
 }
+
+func (p *Provider) ResetCache() {
```
Comment (jonathan-innis) [inline pkg/providers/subnet/provider.go:117]:
```suggestion
func (p *Provider) Reset() {
```

File: pkg/cloudprovider/subnets_test.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ import (
 	. "github.com/aws/karpenter-core/pkg/test/expectations"
 )
 
-var _ = Describe("Subnets", func() {
```
Comment (jonathan-innis) [inline pkg/cloudprovider/subnets_test.go:33]: Can we make sure to move some of these tests that still apply to the cloudprovider to the `suite_test` in the cloudprovider package?

Comment (engedaam) [reply pkg/cloudprovider/subnets_test.go:33]: Moved the remaining tests

File: pkg/cloudprovider/instance.go [hunk 4]
```diff
@@ -38,6 +38,7 @@ import (
 	"github.com/aws/karpenter/pkg/apis/v1alpha1"
 	"github.com/aws/karpenter/pkg/cache"
 	awserrors "github.com/aws/karpenter/pkg/errors"
+	subnet "github.com/aws/karpenter/pkg/providers/subnet"
```
Comment (ellistarn) [inline pkg/cloudprovider/instance.go:41]:
```suggestion
	"github.com/aws/karpenter/pkg/providers/subnet"
```

File: pkg/cloudprovider/instancetypes.go [hunk 5]
```diff
@@ -39,6 +39,7 @@ import (
 	"knative.dev/pkg/logging"
 
 	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	subnet "github.com/aws/karpenter/pkg/providers/subnet"
```
Comment (ellistarn) [inline pkg/cloudprovider/instancetypes.go:42]:
```suggestion
	"github.com/aws/karpenter/pkg/providers/subnet"
```

File: pkg/cloudprovider/suite_test.go [hunk 6]
```diff
@@ -63,6 +63,9 @@ import (
 	machineutil "github.com/aws/karpenter-core/pkg/utils/machine"
 	"github.com/aws/karpenter-core/pkg/utils/pretty"
 	"github.com/aws/karpenter/pkg/fake"
+
+	securitygroup "github.com/aws/karpenter/pkg/providers/securitygroup"
+	subnet "github.com/aws/karpenter/pkg/providers/subnet"
```
Comment (ellistarn) [inline pkg/cloudprovider/suite_test.go:68]:
```suggestion
	"github.com/aws/karpenter/pkg/providers/securitygroup"
	"github.com/aws/karpenter/pkg/providers/subnet"
```

**Review by ellistarn — APPROVED** [review]
Nice!

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**